ANMN/burst_averaged_product/burst_average.py

```diff
@@ -0,0 +1,397 @@
+
+    if instrument_type == 'WQM':
+        output_netcdf_obj.title = 'Burst-averaged biogeochemical measurements at %s' % (input_netcdf_obj.site_code)
+    elif instrument_type == 'CTD':
+        output_netcdf_obj.title = 'Burst-averaged moored CTD measurements at %s' % (input_netcdf_obj.site_code)
+
+    output_netcdf_obj.input_file   = get_input_file_rel_path(input_netcdf_file_path)
+    output_netcdf_obj.history      = 'Created %s' % time.ctime(time.time())
+    output_netcdf_obj.date_created = datetime.now().strftime("%Y-%m-%dT%H:%M:%SZ")
+
+    output_netcdf_obj.geospatial_lat_min = input_netcdf_obj['LATITUDE'][:].min()
+    output_netcdf_obj.geospatial_lat_max = input_netcdf_obj['LATITUDE'][:].max()
+    output_netcdf_obj.geospatial_lon_min = input_netcdf_obj['LONGITUDE'][:].min()
+    output_netcdf_obj.geospatial_lon_max = input_netcdf_obj['LONGITUDE'][:].max()
+
+    output_netcdf_obj.geospatial_vertical_min = min(burst_vars['DEPTH']['var_mean'])
+    output_netcdf_obj.geospatial_vertical_max = max(burst_vars['DEPTH']['var_mean'])
+
+    output_netcdf_obj.time_coverage_start = num2date(time_burst_vals, input_netcdf_obj['TIME'].units, input_netcdf_obj['TIME'].calendar).min().strftime('%Y-%m-%dT%H:%M:%SZ')
+    output_netcdf_obj.time_coverage_end   = num2date(time_burst_vals, input_netcdf_obj['TIME'].units, input_netcdf_obj['TIME'].calendar).max().strftime('%Y-%m-%dT%H:%M:%SZ')
+
+    # set up dimensions
+    output_netcdf_obj.createDimension("TIME", len(time_burst_vals))
+
+    # set up variables
+    var_time = output_netcdf_obj.createVariable("TIME", "d", ("TIME",), fill_value=get_imos_parameter_info('TIME', '_FillValue'))
+    var_lat  = output_netcdf_obj.createVariable("LATITUDE", "d", fill_value=get_imos_parameter_info('LATITUDE', '_FillValue'))
+    var_lon  = output_netcdf_obj.createVariable("LONGITUDE", "d", fill_value=get_imos_parameter_info('LONGITUDE', '_FillValue'))
+
+    for var in burst_vars.keys():
+        fillvalue = get_imos_parameter_info(var, '_FillValue')
+        if fillvalue == []:
+            fillvalue = 999999.0
+
+        output_netcdf_obj.createVariable(var, "f4", ("TIME",), fill_value=fillvalue)
+        output_netcdf_obj.createVariable('%s_burst_min' % var, "f4", ("TIME",), fill_value=fillvalue)
+        output_netcdf_obj.createVariable('%s_burst_max' % var, "f4", ("TIME",), fill_value=fillvalue)
+        output_netcdf_obj.createVariable('%s_burst_sd' % var, "f4", ("TIME",), fill_value=fillvalue)
+        output_netcdf_obj.createVariable('%s_num_obs' % var, "i4", ("TIME",), fill_value=fillvalue)
+
+        # set up 'bonus' var att from original FV01 file to FV02
+        input_var_object   = input_netcdf_obj[var]
+        input_var_list_att = input_var_object.__dict__.keys()
+        var_att_disposable = ['name', 'standard_name', 'long_name', 'valid_min', \
+                              'valid_max', '_FillValue', 'ancillary_variables', \
+                              'axis', 'units', 'ChunkSize']
```

Comment:
@ggalibert My point was that we're producing a new file based on the input file, and we're not applying any unit conversion, so at least the output standard_name and units should be the same as in the input file. If that doesn't match what's in imosParameters.txt, that's too bad. Still better than having input temperature in Kelvin and blindly re-labelling it as Celsius!